core/src/components/segment/segment.tsx

```diff
@@ -431,12 +433,74 @@ export class Segment implements ComponentInterface {
     this.checked = current;
   }
 
+  private getSegmentButton = (selector: 'first' | 'last' | 'next' | 'previous' | 'focused'): HTMLIonSegmentButtonElement | null => {
+    const buttons = this.getButtons().filter(button => !button.disabled);
+    const currIndex = buttons.findIndex(button => button === this.checked || button.classList.contains('ion-focused'));
+    switch (selector) {
+      case 'first':
+        return buttons[0];
+      case 'last':
+        return buttons[buttons.length - 1];
+      case 'next':
+        return buttons[currIndex + 1] || buttons[0];
+      case 'previous':
+        return buttons[currIndex - 1 ] || buttons[buttons.length - 1];
+      case 'focused':
+        return this.el.querySelector('ion-segment-button.ion-focused');
+      default:
+        return null;
+    }
+  }
+
+  private onKeyUp = (ev: KeyboardEvent) => {
+    const isRTL = document.dir === 'rtl';
+    let current;
+    switch (ev.key) {
+      case 'ArrowRight':
+        ev.preventDefault();
+        current = isRTL ? this.getSegmentButton('previous') : this.getSegmentButton('next');
+        break;
+      case 'ArrowLeft':
+        ev.preventDefault();
+        current = isRTL ? this.getSegmentButton('next') : this.getSegmentButton('previous')
+        break;
+      case 'Home':
+        ev.preventDefault();
+        current = this.getSegmentButton('first');
+        break;
+      case 'End':
+        ev.preventDefault();
+        current = this.getSegmentButton('last');
+        break;
+      case ' ':
+      case 'Enter':
+        ev.preventDefault();
+        current = this.getSegmentButton('focused');
+      default:
+        break;
+    }
+
+    const previous = this.checked || current;
+    if (previous && current) {
+      this.checkButton(previous, current);
+    }
+  }
+
+  private handleFallbackFocus() {
+    const buttons = this.getButtons();
+    const noTabButtons = this.getButtons().filter(button => button.tabIndex < 0);
+    if (noTabButtons.length === buttons.length) {
+      buttons[0].tabIndex = 0;
+    }
+  }
+
   render() {
     const mode = getIonMode(this);
     return (
       <Host
         role="tablist"
         onClick={this.onClick}
+        onKeyup={this.onKeyUp}
```

Comment:
Minor style thing, but can we use the `@Listen` decorator here? This is what we do in radio group, and I would like to keep it consistent: https://github.com/ionic-team/ionic-framework/blob/d527414e109c9cc6ae43171911f89a43b2f0691f/core/src/components/radio-group/radio-group.tsx